slsReceiverSoftware/src/DataProcessor.cpp

```diff
@@ -519,6 +501,45 @@ void DataProcessor::PadMissingPackets(char* buf) {
 	}
 }
 
+/** ctb specific */
+void DataProcessor::RearrangeDbitData(char* buf) {
+	int totalSize = (int)(*((uint32_t*)buf));
+	int ctbDigitalDataBytes = totalSize - (*ctbAnalogDataBytes) - (*ctbDbitOffset);
+
+	// no digital data      
+    if (!ctbDigitalDataBytes) {
+        FILE_LOG(logWARNING) << "No digital data for call back, yet dbitlist is not empty.";
+        return;
+	}
+
+	const int numSamples = (ctbDigitalDataBytes / sizeof(uint64_t));
+
+	// ceil as numResult64Bytes could be decimal 
+	const int numResult64Bytes = ceil((double)(numSamples * (*ctbDbitList).size()) / 64.00); 
+	std::vector<uint64_t> result(numResult64Bytes, 0); 
+
+	auto dest = result.data();
+	const int digOffset = FIFO_HEADER_NUMBYTES + sizeof(sls_receiver_header) + (*ctbAnalogDataBytes) + (*ctbDbitOffset);
+	auto source = (uint64_t*)(buf + digOffset);
+
+	// loop through digital bit enable vector
+    for (auto bi : (*ctbDbitList)) {
+		// loop through the frame digital data
+        for (auto ptr = source; ptr < (source + numSamples);) {
+			// extract destination in 64 bit batches
+            for (int i = 0; i != 64; ++i) {
```

Comment:
Should this really be 64 and not the dynamic range?

Reply:
This is what we had decided earlier. The dynamic range for the digital bits in ctb is not known to receiver and will have to be used by the user when processing the file. So, these bits are just put together in 64 bit batches.

Reply:
Ah makes sense!